### ISSUE TITLE
Returned solution is not complete

### ISSUE THREAD

**Issue by Wout4** [MiniZinc/libminizinc#716]
the following model does not give a value for every variable after solving:
```

var bool: BV9;
var bool: BV8;
var bool: BV7;
var bool: BV4;

constraint (BV9) \/ ((not BV8) == (BV7));
constraint ((not BV8) == (BV7)) == (BV4);
constraint BV4;

```
output:
(solver used : gecode 6.3.0)
```
BV9 = false;
BV8 = true;
BV7 = false;
BV4 = BV4;
```

I am assuming this is a problem with CSE, where the variable BV4 get's pruned away, so we do not get a value for it.
This bug does not occur when selecting Coin-BC as the solver

**Comment by Dekker1** [comment]
This issue is resolved by e50d667e5233141956edfc08316bcb181ea75687 (but I forgot to put it in the message)

**Comment by cyderize** [comment]
This should now be fixed in 2.8.0
